temporalio/common.py

```diff
@@ -962,6 +962,47 @@ def set(
 
 MetricMeter.noop = _NoopMetricMeter()
 
+
+@dataclass(frozen=True)
+class Priority:
+    """Priority contains metadata that controls relative ordering of task processing when tasks are
+    backlogged in a queue. Initially, Priority will be used in activity and workflow task queues,
+    which are typically where backlogs exist.
+
+    Priority is (for now) attached to workflows and activities. Activities and child workflows
+    inherit Priority from the workflow that created them, but may override fields when they are
+    started or modified. For each field of a Priority on an activity/workflow, not present or equal
+    to zero/empty string means to inherit the value from the calling workflow, or if there is no
+    calling workflow, then use the default (documented below).
+
+    The overall semantics of Priority are:
+    1. First, consider "priority_key": lower number goes first.
+    (more will be added here later)
+    """
+
+    priority_key: int = 0
+    """Priority key is a positive integer from 1 to n, where smaller integers correspond to higher
+    priorities (tasks run sooner). In general, tasks in a queue should be processed in close to
+    priority order, although small deviations are possible.
+
+    The maximum priority value (minimum priority) is determined by server configuration, and
+    defaults to 5.
+
+    The default priority is (min+max)/2. With the default max of 5 and min of 1, that comes out to
+    3.
+    """
+
+    @staticmethod
+    def from_proto(proto: temporalio.api.common.v1.Priority) -> Priority:
```

Comment:
```suggestion
    def _from_proto(proto: temporalio.api.common.v1.Priority) -> Priority:
```

If `_to_proto` is private, this should be too IMO (I know some structures like `RetryPolicy` we chose to expose both, but I like the idea of not exposing until needed)